deployer/src/lib.rs

```diff
@@ -411,96 +437,78 @@ impl<D: Dal + Sync + 'static> Deployer for DeployerService<D> {
-                        _ => tonic::Status::internal(err.to_string()),
-                    })?;
-            let image_name = container
-                .image_name()
-                .map_err(|err| tonic::Status::internal(err.to_string()))?;
-            let idle_minutes = container.idle_minutes();
-            let is_next = container.is_next();
-
-            let git_metadata = GitMetadata {
-                git_branch: deployment.git_branch,
-                git_commit_hash: deployment.git_commit_hash,
-                git_commit_message: deployment.git_commit_message,
-                git_dirty: deployment.git_dirty,
-            };
-            let runnable_deployment = RunnableDeployment {
-                deployment_id,
-                service_name,
-                service_id,
-                tracing_context: Default::default(),
-                claim,
-                target_ip: None,
-                is_next,
-            };
-
-            self.instate_service(
-                runnable_deployment,
-                git_metadata,
-                image_name,
-                idle_minutes,
-                false,
-            )
-            .await
-            .map_err(|err| {
-                tonic::Status::internal(format!("failed to instate the service: {err}"))
-            })?;
+                    .unwrap_or_else(|err| error!("errored when sending project event: {err}"));
+                }
+
+                while let Some(event) = events_rx.recv().await {
+                    tx.send(Ok(event))
+                        .await
+                        .unwrap_or_else(|err| error!("errored when sending project event: {err}"));
+                }
+            });
+
+            Ok(Response::new(tokio_stream::wrappers::ReceiverStream::new(
+                rx,
+            )))
+        } else {
+            Err(tonic::Status::cancelled(
+                "the events have already been subscribed to. You must unsubscribe first to be able to subscribe again.",
+            ))
         }
+    }
 
-        let new_state = self
-            .dal
-            .service_state(&service_id)
-            .await
-            .map_err(|err| {
-                tonic::Status::internal(format!("failed to fetch the service state {err}"))
-            })?
-            .ok_or(tonic::Status::internal("couldn't find the service state"))?;
-        let new_target_ip = new_state
-            .target_ip(&self.config.network_name)
-            .map_err(|err| {
-                tonic::Status::internal(format!("failed to extract the target ip {err}"))
-            })?;
-
-        Ok(Response::new(TargetIpResponse {
-            target_ip: new_target_ip.to_string(),
-        }))
+    #[instrument(skip_all)]
+    async fn unsubscribe_projects(
+        &self,
+        request: tonic::Request<UnsubscribeProjectsRequest>,
+    ) -> TonicResult<tonic::Response<UnsubscribeProjectsResponse>, tonic::Status> {
+        // request.verify(Scope::ProjectWrite)?;
```

Comment:
Same as above

Reply:
True. Keeping this open because it will have the same solution if we find it. I'll resolve both after we decide what to do.

Reply:
Decided to avoid it for now. The deployer is not on the user network.